src/App.js

```diff
@@ -1,47 +1,66 @@
 import React, { Component } from 'react';
 import Dropdown from './dropdown.jsx';
 import List from './pokeList.jsx';
+// import Loading from './loading.jsx';
+import Loader from 'react-loader-spinner'
 
 class App extends Component {
-  state = { pokedex: [], query: null };
+  state = { pokedex: [], query: null, sortPokemonOrder: 'asc', loading: false };
+
+  // this function grabs the fetched data from the API
+  componentDidMount(){
+    this.fetchData();
+  }
 
   fetchData = async () => {
+    this.setState({ loading: true });
     // grab the API url
     let url = "https://pokedex-alchemy.herokuapp.com/api/pokedex";
-
+    
+    // In order to update the url param, interpolate with choice pokemon
+    // URLsearchParams, including set method
+    const searchOrderParam = new URLSearchParams();
     if (this.state.query) {
-      url = url + `?pokemon=${this.state.query}`
+      searchOrderParam.set('pokemon', this.state.query);
+    }
+    if (this.state.sortPokemonOrder) {
+      searchOrderParam.set('sort', 'pokemon');
+      searchOrderParam.set('direction', this.state.sortPokemonOrder);
     }
+    url = url + `?${searchOrderParam.toString()}`;
 
     // Fetch the Url
     let resonse = await fetch(url);
+    // we use await to avoid a pending promise
     // we need to grab the body object data using json()
     let data = await resonse.json()
-    // we use await to avoid a pending promise
-
+    
     //access the results array within the data
     //set state of the above data + state.
-    this.setState({pokedex: data.results});
-  }
-  // this function grabs the fetched data from the API
-  componentDidMount(){
-      this.fetchData();
+    setTimeout(() => {
+      this.setState({ pokedex: data.results, loading: false });
+    }, 4000)
```

Comment:
You don't actually need to set a timeout for the loading in this case. API calls are naturally asynchronous so if they are in a loading state, your loading animation will appear.